presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisyCountAggregationUtils.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.spi.PrestoException;
+
+import java.security.SecureRandom;
+import java.util.Random;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public class NoisyCountAggregationUtils
+{
+    private NoisyCountAggregationUtils()
+    {
+    }
+
+    public static void updateState(NoisyCountState state, double noiseScale, Long randomSeed)
+    {
+        checkNoiseScale(noiseScale);
+
+        // Update count and retain scale and random seed
+        state.setCount(state.getCount() + 1);
+        state.setNoiseScale(noiseScale);
+
+        if (randomSeed == null) {
+            state.setNullRandomSeed(true);
+        }
+        else {
+            state.setNullRandomSeed(false);
+            state.setRandomSeed(randomSeed);
+        }
+    }
+
+    public static void combineStates(NoisyCountState state, NoisyCountState otherState)
+    {
+        state.setNoiseScale(state.getNoiseScale() > 0 ? state.getNoiseScale() : otherState.getNoiseScale()); // noise scale should be > 0
+        checkNoiseScale(state.getNoiseScale());
+        state.setCount(state.getCount() + otherState.getCount());
+
+        if (state.isNullRandomSeed()) {
+            state.setNullRandomSeed(otherState.isNullRandomSeed());
+            state.setRandomSeed(otherState.getRandomSeed());
```

Comment:
do we need to update the seed if it not null?

Reply:
I am not sure what you meant. If `state` has null random seed, we update it with the random seed of `otherState`. 
If the random seed of `otherState` is also null, then we still have null random seed.

Reply:
The confusion comes from the fact that in updateState you're only updating the seed if the argument is not null, so it looked asymmetric to me

Reply:
I see. We can do it because all non-null seeds have the same value. So if we already have a non-null seed value, there is no need to update it. If our seed is still null, then we try to get it from other state. Hope that make sense